quinn-proto/src/range_set.rs

```diff
@@ -244,6 +271,43 @@ impl<'a> DoubleEndedIterator for EltIter<'a> {
     }
 }
 
+pub struct InsertDrain<'a> {
+    set: &'a mut RangeSet,
```

Comment:
Would be good to have documentation here as to what is the semantic meaning of each of these fields and also probably just the semantics of this type.

Reply:
Good idea.